##1. Cargar el dataset


In [ ]:
import pandas as pd
from sklearn.linear_model import LinearRegression


In [ ]:
url = "https://drive.google.com/uc?id=1q9Agz53BcZwHHpdWUiqHzOMQl-ixMKWC"

# Leer el archivo CSV
df = pd.read_csv(url)

# Ver las primeras filas
print(df)

   horas_estudio  asistencia  nota_matematicas
0              5          80                70
1             10          90                85
2              2          70                50
3              8          85                78
4             12          95                92
5              3          60                55
6              7          75                75
7              9          90                82
8              4          80                65
9             11          95                90


##2. Preparar los datos

In [ ]:
## Preparamos las variables tanto independientes como dependientes
# Variables independientes (horas de estudio, asistencia)
X = df[['horas_estudio', 'asistencia']]

# Variable dependiente (nota de matemáticas)
y = df['nota_matematicas']


##3. Crear y entrenar el modelo

In [ ]:
# Crear el modelo de regresión lineal
model = LinearRegression()

# Entrenar el modelo
model.fit(X, y)


LinearRegression()

##4. Realizar predicciones

In [ ]:
# Realizar predicción
nota_predicha = model.predict([[13, 85]])
print("Nota predicha:", nota_predicha[0])


Nota predicha: 94.68640873015875


/usr/local/lib/python3.10/dist-packages/sklearn/base.py:493: UserWarning: X does not have valid feature names, but LinearRegression was fitted with feature names
  warnings.warn(


##5. Evaluar el modelo

In [ ]:
from sklearn.metrics import r2_score

# Predecir con el modelo entrenado para los datos de entrenamiento
y_pred = model.predict(X)

# Calcular el R-cuadrado
r2 = r2_score(y, y_pred)
print("R-cuadrado:", r2)


R-cuadrado: 0.9759802719464814


##6. Explicar el modelo

¿Qué indica el intercepto del modelo? ¿Cuál de las variables predictoras tiene una mayor influencia en la nota de matemáticas? ¿Qué otras variables podrían ser relevantes para predecir el desempeño en matemáticas? ¿Cómo podrían incorporarse al modelo?

In [ ]:
"""
DESARROLLO DE LAS PREGUNTAS

1. Intercepto del Modelo: El intercepto representa la nota de matemáticas esperada cuando tanto las horas de estudio como la asistencia son cero.
"""
## 2.
print("Intercepto:", model.intercept_)
print("Coeficientes:", model.coef_)


Intercepto: 31.736507936507948
Coeficientes: [3.35674603 0.22720238]


2. Con el intercepto y los coeficientes podemos interpretar la influencia de cada variable predictora en el modelo.

Intercepto (31.736): Representa la nota de matemáticas predicha cuando tanto las horas de estudio como la asistencia son cero. En este contexto, indica una base de rendimiento (31.736 puntos) sin horas de estudio ni asistencia. Es decir, el modelo predice que un estudiante que no estudia ni asiste a clases obtendría esta nota.

Influencia de las Variables Predictoras:

Horas de estudio (3.36): Por cada hora adicional de estudio semanal, se espera que la nota de matemáticas aumente en aproximadamente 3.36 puntos, manteniendo la asistencia constante. Esto sugiere que las horas de estudio tienen una influencia significativa en la mejora del rendimiento.

Asistencia (0.23): Por cada 1% adicional de asistencia, la nota de matemáticas aumenta en aproximadamente 0.23 puntos, manteniendo constantes las horas de estudio. Aunque la asistencia también contribuye positivamente, su influencia en la nota es menor en comparación con las horas de estudio.
Por lo tanto, las horas de estudio tienen una mayor influencia en la nota de matemáticas, según los coeficientes obtenidos.

3. Otras Variables Relevantes: Podrían incluirse variables como:
Participación en actividades extracurriculares, Ambiente de estudio en casa, Nivel socioeconómico.

Cómo Incorporarlas: Recolectar estos datos y agregarlos como nuevas columnas en el dataset para incluirlos en el conjunto de variables predictoras. Esto permitiría al modelo tomar en cuenta más factores que influyen en el desempeño.